### PR TITLE
Generic/PHP/Syntax: add XML documentation

### PR DESCRIPTION
**Pull request by rodrigoprimo**
## Description
This PR adds documentation for the Generic.PHP.Syntax sniff.

I'm unsure about what to use in the `title` attribute. I followed the convention and used the sniff name, but just `Syntax` is not very clear. Happy to leave it as is if that is ok. An alternative that I considered using is `PHP Syntax`.

## Suggested changelog entry
Add documentation for the Generic.PHP.Syntax sniff

## Related issues/external references

Part of #148

## Types of changes

- [ ] Bug fix _(non-breaking change which fixes an issue)_
- [ ] New feature _(non-breaking change which adds functionality)_
- [ ] Breaking change _(fix or feature that would cause existing functionality to change)_
    - [ ] This change is only breaking for integrators, not for external standards or end-users.
- [x] Documentation improvement

## PR checklist

- [x] I have checked there is no other PR open for the same change.
- [x] I have read the [Contribution Guidelines](https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/.github/CONTRIBUTING.md).
- [x] I grant the project the right to include and distribute the code under the BSD-3-Clause license (and I have the right to grant these rights).
- [ ] I have added tests to cover my changes.
- [x] I have verified that the code complies with the projects coding standards.
- [ ] [Required for new sniffs] I have added XML documentation for the sniff.

### CODE REVIEW

File: src/Standards/Generic/Docs/PHP/SyntaxStandard.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        <code title="Valid: No PHP syntax errors.">
+        <![CDATA[
+<em>echo "Hello!";</em>
+<em>$array = [1, 2, 3];</em>
+        ]]>
+        </code>
+        <code title="Invalid: Code contains PHP syntax errors.">
+        <![CDATA[
+echo "Hello!"<em></em> // Missing semicolon at the end
+$array = [1, 2, 3<em></em>; // Missing closing bracket
```
Comment (jrfnl) [inline src/Standards/Generic/Docs/PHP/SyntaxStandard.xml:17]: Tiny nitpick.

```suggestion
echo "Hello!"<em></em> // Missing semicolon.
$array = [1, 2, 3<em></em>; // Missing closing bracket.
```

**Review by jrfnl — COMMENTED** [review]
@rodrigoprimo Thanks for this PR. Looking good.

> I'm unsure about what to use in the `title` attribute. I followed the convention and used the sniff name, but just `Syntax` is not very clear. Happy to leave it as is if that is ok. An alternative that I considered using is `PHP Syntax`.

I 100% agree with you. `PHP Syntax` would be better.

**Comment by rodrigoprimo** [comment]
Thanks for the review, @jrfnl.

I updated the `title` to `PHP Syntax` and made the adjustments you suggested in the inline comments (https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/175#discussion_r1436307753).

I amended the original commit instead of creating new ones per your comment in a previous PR: https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/163#pullrequestreview-1779920437. Please let me know if, in this case, you would have preferred new commits.

This PR is ready for another review.

**Review by jrfnl — APPROVED** [review]
Thanks for making those changes @rodrigoprimo ! LGTM.

> I amended the original commit instead of creating new ones per your comment in a previous PR: [#163 (review)](https://github.com/PHPCSStandards/PHP_CodeSniffer/pull/163#pullrequestreview-1779920437). Please let me know if, in this case, you would have preferred new commits.

As a general rule of thumb:
* For very small punctuation/doc fixes, amending is fine to keep the history clean.
* For more significant/functional changes, a separate commit is preferred as that allows for a reviewer to see more easily what has been changed in the update.

When in doubt, use an extra commit (and I can then decide whether to squash the commits on merge or leave it as-is).

**Pull request merged**